app/services/hyrax/file_set_audit_service.rb

```diff
@@ -60,8 +60,11 @@ def stat_to_string(stat)
       # @param [ActiveFedora::File] file to audit
       # @param [Array] log container for messages
       def audit_file(file, log = [])
-        versions = file.has_versions? ? file.versions.all : file
-        versions.each { |v| log << audit_file_version(file.id, v.uri) }
+        if file.has_versions?
```

Comment:
@bess What would you think of simplifying the code changes made in this branch to better align with the related code in `master` (which should not be vulnerable to the same buggy behavior):

https://github.com/samvera/hyrax/blob/master/app/services/hyrax/file_set_fixity_check_service.rb#L79

You could patch line 63 as follows:

```ruby
version = file.has_versions?  file.versions.all : Array.wrap(file)
```

As for how to test it, take a look at this related code in `master`:

https://github.com/samvera/hyrax/blob/master/spec/services/hyrax/file_set_fixity_check_service_spec.rb#L10-L28

Happy to chat more about this, and have time tomorrow. Thanks for working on this!